plugins/woocommerce/src/Blocks/OptionsManager.php

```diff
@@ -0,0 +1,37 @@
+<?php
+namespace Automattic\WooCommerce\Blocks;
+
+use Automattic\WooCommerce\Blocks\Options;
+
+/**
+ * OptionsManager class.
+ *
+ * @internal
+ */
+class OptionsManager {
+
+	/**
+	 * Initialization method.
+	 */
+	public function init() {
+		add_action( 'after_switch_theme', array( $this, 'check_should_use_blockified_product_grid_templates' ), 10, 2 );
+	}
+
+	/**
+	 * Checks the old and current themes and determines if the "wc_blocks_use_blockified_product_grid_block_as_template"
+	 * option need to be updated accordingly.
+	 *
+	 * @return void
+	 */
+	public function check_should_use_blockified_product_grid_templates() {
```

Comment:
This code feels like it should belong closer to the templating logic. I'd expect an `OptionsManager` class to provide me with generic methods, that would be extended. Whereas this method feels very specific to a single part of functionality.

Reply:
We intentionally moved this code out from `BlockTemplatesController` because it needs to run when `BlockTemplatesController` is not initialized. The main goal is to make sure `BlockTemplatesController` is not initialized in classic themes.

Would renaming this class from `OptionsManager` to something more specific help? Ie: `TemplateOptions` or something along these lines? :thinking:

Reply:
I changed the class name to `TemplateOptions` in bedc8157fde8ed210eaa33b24b59fcef82447f24.